app/models/amusement_park.rb

```diff
@@ -1,3 +1,9 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :ride_mechanics, through: :rides
+  has_many :mechanics, through: :ride_mechanics
+
+ def park_mechanics
+  self.mechanics.select(:name).distinct
```

Comment:
Great! If you need the whole mechanics object, you can also leave the `select` out (for this US it doesn't matter either way)

Reply:
Because this is encapsulated into a model method (great MVC implementation!) we do want to see a model test for this functionality to verify that the right mechanics are returned when this is called on an amusement park.